File: willow-api/api/src/main/groovy/ish/oncourse/willow/service/SearchApi.java
package ish.oncourse.willow.service;

import ish.oncourse.willow.model.common.CommonError;
import ish.oncourse.willow.model.common.Item;

import java.io.InputStream;
import java.io.OutputStream;
import java.util.List;
import java.util.Map;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.MediaType;

@Path("/")
public interface SearchApi  {

    @GET
    @Path("/country/{text}")
    @Produces({ "application/json" })
    List<Item> getCountries(@PathParam("text") String text);

    @GET
    @Path("/language/{text}")
    @Produces({ "application/json" })
    List<Item> getLanguages(@PathParam("text") String text);

    @GET
    @Path("/postcode/{text}")
    @Produces({ "application/json" })
    List<Item> getPostcodes(@PathParam("text") String text);

    @GET
    @Path("/suburb/{text}")
    @Produces({ "application/json" })
    List<Item> getSuburbs(@PathParam("text") String text);
}

File: checkout/src/js/http/SearchApi.ts
import {HttpService} from "../common/services/HttpService";
import {CommonError} from "../model/common/CommonError";
import {Item} from "../model/common/Item";

export class SearchApi {
  constructor(private http: HttpService) {
  }

  getCountries(text: string): Promise<Item[]> {
    return this.http.GET(`/v1/country/${text}`);
  }
  getLanguages(text: string): Promise<Item[]> {
    return this.http.GET(`/v1/language/${text}`);
  }
  getPostcodes(text: string): Promise<Item[]> {
    return this.http.GET(`/v1/postcode/${text}`);
  }
  getSuburbs(text: string): Promise<Item[]> {
    return this.http.GET(`/v1/suburb/${text}`);
  }
}
